modules/distributor/distributor.go

```diff
@@ -347,3 +355,17 @@ func requestsByTraceID(req *tempopb.PushRequest, userID string, spanCount int) (
 
 	return keys, pushRequests, nil
 }
+
+func recordDiscaredSpans(err error, userID string, spanCount int) {
+	s := status.Convert(err)
+	if s == nil {
+		return
+	}
+	desc := s.Message()
+
+	if strings.HasPrefix(desc, overrides.ErrorPrefixLiveTracesExceeded) {
```

Comment:
nit: Metric'ing based on an error message prefix from a different service seems fragile.. We could wrap the error or change it in many different ways in the future. Should we change this to a `strings.Contains()`? Or should we map different codes in the `codes` package to our errors?

Reply:
> Metric'ing based on an error message prefix from a different service seems fragile..
We have tests to detect drift.  I prefer prefix to contains as it's more performant and during an event with tons of errors we would want to this code path to be as efficient as possible. 

> Or should we map different codes in the codes package to our errors?
We need to return specific and meaningful GRPC calls so downstream will react appropriately.  Also if we hijack different grpc status codes to mean different failures due to limits we'll run out eventually.  We definitely need something more robust.

100% agree that it's not great, but I think the string error is the best way to pass custom error information back through a GRPC rpcerror.

Reply:
If nested errors are preserved across the gRPC boundary we could use `errors.Is` and const values like `errLiveTracesExceeded`

Reply:
Sadly, it is not.  We had the same issues with the Jaeger GRPC plugin and decided to go with string matching.

Reply:
In that case it seems like string prefix is the best option.